### ISSUE TITLE
Keyboard announced as hidden when changing text field content using TextEditingController

### ISSUE THREAD

**Issue by alanrussian** [flutter/flutter#31796]
Repro steps:
1) Build and run the code below.
2) Turn on TalkBack on Android.
2) Focus on the text field and open it by double tapping.
3) Focus and double tap on the "PRESS ME" button.

Expected: Not entirely sure... Perhaps TalkBack should announce the new value?

Observed: "Showing English QWERTY keyboard. English US QWERTY keyboard hidden."

This is an accessibility issue because the keyboard is not actually hidden. It would be very confusing to a person who cannot see. 

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  final String title;

  const MyHomePage({Key key, this.title}) : super(key: key);

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return MyScaffold();
  }
}

class MyScaffold extends StatelessWidget {
  const MyScaffold({Key key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: SingleChildScrollView(
          child: TestWidget(),
        ),
      ),
    );
  }
}

class TestWidget extends StatefulWidget {
  @override
  _TestWidgetState createState() => _TestWidgetState();
}

class _TestWidgetState extends State<TestWidget> {
  final _controller = TextEditingController();
  var i = 0;

  @override
  Widget build(BuildContext context) {
    return Column(
      children: <Widget>[
        TextField(
          controller: _controller,
        ),
        FlatButton(
          onPressed: () {
            _controller.text = '${_controller.text}$i';
          },
          child: Text('PRESS ME'),
        )
      ],
    );
  }
}
```

**Comment by alanrussian** [comment]
cc @goderbauer

**Comment by HansMuller** [comment]
Thanks for the test case, the problem is easy to reproduce. Looking into it.

**Comment by HansMuller** [comment]
This appears to a native Android bug. A test case based on the following code fails in the same way:
```
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final EditText textField = findViewById(R.id.editText);
        final Button button = findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                textField.setText(textField.getText() + "0");
            }
        });

    }
}
```

**Comment by HansMuller** [comment]
Filed an Android bug: b/131858638

**Comment by Hixie** [comment]
Status: We are blocked awaiting upstream fixes.

**Comment by HansMuller** [comment]
Just FTR: I believe the version of the Android SDK I was using when I reported this bug was 28.0.3

**Comment by mehmetf** [comment]
Is this a blocker or critical? The internal Android bug is marked as P2.

**Comment by Hixie** [comment]
@mehmetf It was marked as blocking customer:mulligan by @alanrussian .

**Comment by darshankawar** [comment]
Hi @alanrussian,
I tried to replicate this issue on Android 8 and observed that talkback doesn't announce `Showing English QWERTY keyboard. English US QWERTY keyboard hidden."` or anything similar that will indicate that keyboard is hidden. 

The keyboard remains open and when I typed on it, the corresponding letters were read properly.
I tested it on latest Flutter dev channel.

Can you please confirm if you still experience this issue on latest Flutter version ?

**Comment by darshankawar** [comment]
Without additional information we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
